Few fixes to get `asyncipc` working with devel on windows

ire4ever1190 opened the pull request:
Closes #43  

 - First fix was to make sure the object is `WideCString` first before casting to pointer so that we are actually accessing the data (On devel this calls a converter which returns the internal data)

 - I was running into a nil access error on ORC which I fixed by using the stdlib `getGlobalDispatcher()` instead

ire4ever1190 committed:
Multiple fixes to get asyncipc working with devel on windows

 - First was to make sure the object is  first before casting to pointer so that we are actually accessing the data (On devel this calls a converter which returns the type

 - I was running into a nil access error on ORC which I fixed by using the stdlib getGlobalDispatcher() instead

ire4ever1190 commented on asynctools/asyncipc.nim:
```diff
@@ -267,7 +266,7 @@ elif defined(windows):
 
     if register:
-      let p = getCurrentDispatcher()
+      let p = getGlobalDispatcher()
```
From what I can gather the use of `getCurrentDispatcher` was to access the `ioPort` handle? 1.6 (atleast I think 1.6) introduced [privateAccess](https://nim-lang.org/docs/importutils.html#privateAccess%2Ctypedesc) which allows us to get that handle without needing `HackDispatcher`.

If thats correct, should I make a follow up PR to replace `getCurrentDispatcher` with `getGlobalDispatcher`? (or even just do it in this PR)

Araq replied:
Oh god no. Don't use `privateAccess` to hack your way into the standard library. Instead export the `ioPort` handle or an accessor proc.

ire4ever1190 replied:
Found [getIoHandler](https://nim-lang.org/docs/asyncdispatch.html#getIoHandler%2CPDispatcher) so no need in the end.

Does make me wonder what the point of `HackDispatcher` is then?

ire4ever1190 committed:
Still use HackDispatcher when using the old runtime since it doesn't have any problems there

This is because we need to access the handles which wasn't exported til recently so 1.2.12 needs the hack. But if you are using the new runtime then you are likely using a newer version of Nim

Araq commented on asynctools/asyncipc.nim:
```diff
@@ -130,6 +130,7 @@ elif defined(windows):
       ioPort: Handle
       handles: HashSet[AsyncFD]
     HackDispatcher = ptr HackDispatcherImpl
+    Dispatcher = HackDispatcher | PDispatcher
```
Too hacky and not required.

Araq commented on asynctools/asyncipc.nim:
```diff
@@ -322,7 +336,7 @@ elif defined(windows):
   template registerWaitableChange(ipc: AsyncIpcHandle, pcd, handleCallback) =
     let p = getCurrentDispatcher()
```
```suggestion
    when declared(getIoHander):
      pcd.ioPort = p.getIoHandler()
    else:
      pcd.ioPort = cast[ptr Thing](p).ioPort # direct access
```

ire4ever1190 committed:
Just define getIoHandler for HackDispatcher instead of using when

Simplifies it and feels less hacky

Remove the type that Araq says sn't required
